app/actions/websocket/users.ts

```diff
@@ -22,49 +22,60 @@ import type UserModel from '@typings/database/models/servers/user';
 const {SERVER: {CHANNEL, CHANNEL_MEMBERSHIP}} = MM_TABLES;
 
 export async function handleUserUpdatedEvent(serverUrl: string, msg: any) {
-    const database = DatabaseManager.serverDatabases[serverUrl];
-    if (!database) {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
         return;
     }
-    const currentUser = await queryCurrentUser(database.database);
+    const currentUser = await queryCurrentUser(operator.database);
     if (!currentUser) {
         return;
     }
 
     const user: UserProfile = msg.data.user;
+    const modelsToBatch: Model[] = [];
+    let userToSave = user;
 
     if (user.id === currentUser.id) {
         if (user.update_at > (currentUser?.updateAt || 0)) {
             // Need to request me to make sure we don't override with sanitized fields from the
             // websocket event
-            // TODO Potential improvement https://mattermost.atlassian.net/browse/MM-40582
-            fetchMe(serverUrl, false);
+            if (!user.notify_props || !Object.keys(user.notify_props).length) {
+                // Current user is sanitized so we fetch it from the server
+                const me = await fetchMe(serverUrl, true);
+                if (me.user) {
+                    userToSave = me.user;
+                }
+            }
 
             // Update GMs display name if locale has changed
             if (user.locale !== currentUser.locale) {
-                const channels = await database.database.get<ChannelModel>(CHANNEL).query(
+                const channels = await operator.database.get<ChannelModel>(CHANNEL).query(
                     Q.where('type', Q.eq(General.GM_CHANNEL))).fetch();
-                const {models} = await updateChannelsDisplayName(serverUrl, channels, user, true);
+                const {models} = await updateChannelsDisplayName(serverUrl, channels, [user], true);
```

Comment:
On line 64 we check for the length of the channels, but we don't here. Not sure if relevant to check or not, but would be nice to be consistent.